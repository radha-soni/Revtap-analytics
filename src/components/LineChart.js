import React, { useEffect } from "react";
import { Line } from "react-chartjs-2";
import Paper from "@material-ui/core/Paper";
import "./chart.css";

export default function LineChart({ order }) {
  const created = [];
  const prices = [];

  if (order) {
    order.forEach((element) => {
      created.push(element.created.slice(0, 10));
    });
    created.sort();
    removeDuplicate(created);
  }

  function removeDuplicate(created) {
    created.splice(0, created.length, ...new Set(created));
    created.forEach((element) => {
      let value = 0;
      order.forEach((date) => {
        if (element === date.created.slice(0, 10)) {
          value = value + Number(date.price);
        }
      });
      prices.push(value);
    });
  }
  const data = {
    labels: created,
    datasets: [
      {
        fill: false,
        lineTension: 0,
        backgroundColor: "rgba(75,192,192,1)",
        borderColor: "#1976d2",
        borderWidth: 2,
        pointStyle: "circle",
        radius: 1,
        data: prices,
      },
    ],
  };

  const options = {
    scales: {
      xAxes: [
        {
          type: "time",
          gridLines: {
            display: false,
          },
        },
      ],
    },
    legend: {
      display: false,
    },
    maintainAspectRatio: false,
    responsive: false,
  };

  return (
    <Paper elevation={3} className="paper">
      <div>
        <p>Orders Total Price</p>
      </div>
      <Line data={data} height={400} width={600} options={options} />
    </Paper>
  );
}
