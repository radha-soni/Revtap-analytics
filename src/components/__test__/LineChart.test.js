import React from "react";
import renderer from "react-test-renderer";
import LineChart from "../LineChart";
test("line chart renders correctly", () => {
  const order = [
    {
      id: "3eec09ba-92f3-4f3e-abe4-5775decca863",
      created: "2020-04-13T12:19:08.890Z",
      customer: "Hills",
      product: "Gloves",
      price: "833.00",
    },
  ];

  const chart = renderer.create(<LineChart order={order} />).toJSON();

  expect(chart).toMatchSnapshot();
});
