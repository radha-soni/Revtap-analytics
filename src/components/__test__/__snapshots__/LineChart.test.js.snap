// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`line chart renders correctly 1`] = `
<div
  className="MuiPaper-root paper MuiPaper-elevation3 MuiPaper-rounded"
>
  <div>
    <p>
      Orders Total Price
    </p>
  </div>
</div>
`;
